test_specific_improvements.py

```
@@ -0,0 +1,394 @@
+    # Test 2: Empty file detection
+    with tempfile.NamedTemporaryFile(suffix='.png', delete=False) as tmp:
+        empty_path = Path(tmp.name)
+    
+    try:
+        original_argv = sys.argv
+        sys.argv = ['pipeline.py', str(empty_path)]
+        
+        try:
+            with patch('builtins.print') as mock_print:
+                pipeline.main()
+            assert False, "Should have detected empty file"
```

Ruff (0.11.9)

Do not `assert False` (`python -O` removes these calls), raise `AssertionError()`

Replace `assert False`

(B011)